### ISSUE TITLE
Mongoose v6 modifies connection options object

### ISSUE THREAD

**Issue by BrettHoutz** [Automattic/mongoose#14370]
### Prerequisites

- [x] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.12.6

### Node.js version

20.10.0

### MongoDB server version

6.0.4

### Description

`mongoose.connect()` sets the properties `config`, `autoCreate`, and `autoIndex`, which i understand to be mongoose-specific connection options. My application creates connections using both `mongoose` and `mongodb` and uses a common options object for both. When the options object is passed to `mongodb.MongoClient.connect()`, it has the properties that were added by mongoose, which causes an error from MongoClient.

### Steps to Reproduce

```js
const mongoose = require('mongoose');

const options = {
  connectTimeoutMS: 5000,
  socketTimeoutMS: 30000,
  readPreference: 'nearest',
};

mongoose.connect('mongodb://localhost:27017', options).then(() => {
  // config, autoCreate, and autoIndex are set in options
  console.log(options);
});
```

### Expected Behavior

Since the options object is a set of static properties for configuring the behavior of Mongoose, i would expect that it not be modified at all.

**Comment by donoftime2018** [comment]
What I did was update mongoose to latest version 8.2.0. And it only shows the options listed in the options object.

```
#!/usr/bin/env node

const mongoose = require('mongoose');

const options = {
  connectTimeoutMS: 5000,
  socketTimeoutMS: 30000,
  readPreference: 'nearest'
};

mongoose.connect('mongodb://localhost:27017', options).then(() => {
  console.log(options); //latest version of mongoose only shows the options listed
});
```

**Comment by donoftime2018** [comment]
If you want to keep it at mongoose version 6.12.6, then removing readPreference only shows the fields listed in options.

```
#!/usr/bin/env node

const mongoose = require('mongoose');

const options = {
  connectTimeoutMS: 10,
  socketTimeoutMS: 10,
  autoCreate: false,
  autoIndex: true,
  config: {autoIndex: true},
//   readPreference: 'nearest'
};

mongoose.connect('mongodb://localhost:27017', options).then(() => {
  console.log(options); //latest version of mongoose only shows the options listed
});
```

**Comment by donoftime2018** [comment]
> What I did was update mongoose to latest version 8.2.0. And it only shows the options listed in the options object.
> 
> ```
> #!/usr/bin/env node
> 
> const mongoose = require('mongoose');
> 
> const options = {
>   connectTimeoutMS: 5000,
>   socketTimeoutMS: 30000,
>   readPreference: 'nearest'
> };
> 
> mongoose.connect('mongodb://localhost:27017', options).then(() => {
>   console.log(options); //latest version of mongoose only shows the options listed
> });
> ```

I understand you wanted to not set other fields only on mongoose v6.

**Comment by vkarpov15** [comment]
This is a duplicate of #13335, fixed by cherry picking #13376.
